src/js/hooks/useInvoiceFilters.jsx

```diff
@@ -0,0 +1,122 @@
+      const response = await apiClient.get(`/openboxes/api/generic/person/${id}`);
+      return response.data?.data;
+    } catch (e) {
+      Alert.error(translate(
+        'react.invoice.error.userFetching.label',
+        'Could not fetch user for createdBy filter',
+      ));
+    }
+    return null;
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      // Avoid unnecessary re-fetches if getAppContext triggers fetching session info
+      // but currentLocation doesn't change
+      if (currentLocation?.id) {
+        const initialEmptyValues = Object.keys(filterFields).reduce((acc, key) => {
+          if (!acc[key]) return { ...acc, [key]: '' };
+          return acc;
+        }, {});
+        const queryProps = queryString.parse(history.location.search);
+
+        // IF VALUE IS IN A SEARCH QUERY SET DEFAULT VALUES
+        if (queryProps.status) {
+          initialEmptyValues.status = statuses
+            .find(({ value }) => value === queryProps.status);
+        }
+        if (queryProps.vendor) {
+          initialEmptyValues.vendor = suppliers.find(({ value }) => value === queryProps.vendor);
+        }
+        if (queryProps.invoiceTypeCode) {
+          initialEmptyValues.invoiceTypeCode =
+            typeCodes.find(({ value }) => value === queryProps.invoiceTypeCode);
+        }
+        if (queryProps.dateInvoiced) {
+          initialEmptyValues.dateInvoiced = queryProps.dateInvoiced;
+        }
+        if (queryProps.createdBy) {
+          initialEmptyValues.createdBy = queryProps.createdBy === currentUser?.id
+            ? currentUser
+            : await fetchUserById(queryProps.createdBy);
+        }
+
+        setDefaultValues({
+          ...initialEmptyValues,
+          buyerOrganization: {
+            id: currentLocation?.organization?.id,
+            value: currentLocation?.organization?.id,
+            name: currentLocation?.organization?.name,
+            label: currentLocation?.organization?.name,
+          },
+        });
+      }
+    })();
+  }, [currentLocation?.id]);
+
+  useEffect(() => {
+    // If statuses or invoice type codes not yet in store, fetch them
+    if (!statuses || !statuses.length) {
+      dispatch(fetchInvoiceStatuses());
+    }
+    if (!typeCodes || !typeCodes.length) {
+      dispatch(fetchInvoiceTypeCodes());
+    }
+
+    // TODO: If editing organizations is in React,
+    //  fetch only if length === 0, as edit would should force refetch anyway
+    dispatch(fetchSuppliers());
+  }, []);
+
+  const setFilterValues = useCallback((values) => {
+    const filterAccessors = {
+      buyerOrganization: { name: 'buyerOrganization', accessor: 'id' },
+      status: { name: 'status', accessor: 'id' },
+      vendor: { name: 'vendor', accessor: 'id' },
+      invoiceTypeCode: { name: 'invoiceTypeCode', accessor: 'id' },
+      dateInvoiced: { name: 'dateInvoiced' },
+      createdBy: { name: 'createdBy', accessor: 'id' },
+    };
+
+    if (Object.keys(values).length > 0) {
```

Comment:
`Object.keys(values).length` is always equal or higher than 0, so you can simplify this condition to: `Object.keys(values).length`. Numbers higher than 0 are truthy, only 0 is falsy.

Reply:
@alannadolny You are obviously right, I just didn't catch it when copying this part of code, as it was already implemented - funny thing is that in lines `88` and `91` I caught similar issue and remembered to change
from:
```
if (!statuses || statuses.length === 0)
```
to:
```
if (!statuses || !statuses.length)
```